diff --git a/app/code/core/Mage/Customer/Model/Address/Abstract.php b/app/code/core/Mage/Customer/Model/Address/Abstract.php
index 5d0a577..fc42e47 100644
--- a/app/code/core/Mage/Customer/Model/Address/Abstract.php
+++ b/app/code/core/Mage/Customer/Model/Address/Abstract.php
@@ -367,7 +367,9 @@ class Mage_Customer_Model_Address_Abstract extends Mage_Core_Model_Abstract
 
         $this->_basicCheck();
 
-        Mage::dispatchEvent('customer_address_validation_after', array('address' => $this));
+        if (!$this->_getErrors()) {
+            Mage::dispatchEvent('customer_address_validation_after', array('address' => $this));
+        }
 
         $errors = $this->_getErrors();
 
